llama-index-core/llama_index/core/query_pipeline/query.py

```diff
@@ -352,10 +363,27 @@ def run_with_intermediates(
                     **kwargs,
                 )
 
+    def merge_dicts(self, d1, d2):
```

Comment:
I'm 99% sure it does, but I guess it's important to maintain order when calling this (i.e. we need to return results in the same order as the batches)

Reply:
Yeah, merge_dicts maintains the order